server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -696,6 +710,7 @@ public boolean freeContext(long id) {
                 assert context.refCount() > 0 : " refCount must be > 0: " + context.refCount();
                 context.indexShard().getSearchOperationListener().onFreeContext(context);
                 if (context.scrollContext() != null) {
+                    openScrollContexts.decrementAndGet();
```

Comment:
we should also check the scroll contexts in `freeAllContextForIndex` above ?

Reply:
Doesn't `freeAllContextForIndex` simply call `freeContext` ? Am I supposed to do any extra processing here?

Reply:
Sorry my bad, it does call `freeContext`